crates/project_panel/src/project_panel.rs

```diff
@@ -82,7 +85,7 @@ pub struct ProjectPanel {
     show_scrollbar: bool,
     scrollbar_drag_thumb_offset: Rc<Cell<Option<f32>>>,
     hide_scrollbar_task: Option<Task<()>>,
-    diagnostic_error_paths: HashSet<PathBuf>,
+    diagnostics: HashMap<(WorktreeId, PathBuf), DiagnosticSeverity>,
```

Comment:
@mrnugget I went for a tuple to avoid chaining HashMaps if that is ok for you?

Reply:
Absolutely, good idea.